### DISCUSSION TITLE
A small curiosity re login process

### DISCUSSION THREAD

**Discussion by cerebrate** [microsoft/WSL#8926] [General]
I've been noticing this login process here:

![image](https://user-images.githubusercontent.com/371623/193095264-cd8f786c-aa23-40d6-83b1-066a10e7516e.png)

As that doesn't seem to be one of mine... what does that do?

**Comment by NotTheDr01ds** [comment]
Still seeing this?

On 22.04, (and I thought 20.04), `systemd-container` (which provides `machinectl`) isn't installed by default.  And even after installing it and restarting (on a fresh install of 22.04, at least), I don't see that process with Systemd enabled.

**Reply by cerebrate** [reply]
Oh, no, the machinectl _is_ mine. My _bottle-imp_ software, like _genie_, uses _machinectl_ to create a user session that _sd-pam_/_systemd-logind_ recognizes, as that's necessary in order to get the user session/user systemd/session dbus set up properly. (Per #8842 etal.)

It's the _login_ and _zsh_ under the top-level _init_ I'm wondering about, which appear to be created automatically by WSL.

**Comment by NotTheDr01ds** [comment]
Ah, sorry - I see now.  I was thinking the `zsh`/`login` was a subprocess of the `machinectl`, but now I see that it descends from the Systemd-initiated WSL `init`.

And yes, I do see the same `login -f` with a `bash` on mine.  I thought when I saw your post that I'd seen something similar, but then I went off on the (personal error) red-herring of looking for `machinectl`.
